synthesizer/process/src/stack/authorization/mod.rs

```diff
@@ -147,6 +148,73 @@ impl<N: Network> Authorization<N> {
             _ => false,
         }
     }
+
+    /// Checks whether the authorization is for a valid program edition.
+    pub fn check_valid_edition(&self, process: &crate::Process<N>, _consensus_version: ConsensusVersion) -> Result<()> {
+        // Determine the root transition's program ID.
+        let transitions = self.transitions.read();
+        let root_program_id = *transitions
+            .first()
+            .map(|(_, t)| t.program_id())
+            .ok_or_else(|| anyhow!("No transitions found in the Authorization."))?;
+        // There is only one credits.aleo edition, so we can safely skip this case.
+        if root_program_id.to_string() != "credits.aleo" {
+            // Get the program's current edition.
+            let _program_edition = *process.get_stack(root_program_id)?.program_edition();
+            // If we're past ConsensusVersion::V8, ensure new stacks are not on edition 0.
+            #[cfg(not(any(test, feature = "test")))]
+            if _consensus_version >= ConsensusVersion::V8 && _program_edition == 0 {
```

Comment:
Please add documentation here so we are aware of this check when Upgradability will land.

Once upgradability lands, this check will be different. We can't just check the program edition anymore, it will need to be programs that are edition 0 with no constructor that can't be called.

Reply:
635bbf6bf